# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/Object.h
# Commit: 22854b710fbf
# Full Hash: 22854b710fbfb779ac6b19fe4c9ac2570189cec0
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 6 - Move Realm* from ObjectGroup to BaseShape. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106976
# ==============================================================================

diff -r 4531d5c25694 -r 22854b710fbf js/public/Object.h
--- a/js/public/Object.h	Fri Mar 05 19:10:06 2021 +0000
+++ b/js/public/Object.h	Fri Mar 05 19:10:07 2021 +0000
@@ -56,7 +56,7 @@
  * compartment of this realm.
  */
 static MOZ_ALWAYS_INLINE Compartment* GetCompartment(JSObject* obj) {
-  Realm* realm = reinterpret_cast<shadow::Object*>(obj)->group->realm;
+  Realm* realm = reinterpret_cast<shadow::Object*>(obj)->shape->base->realm;
   return GetCompartmentForRealm(realm);
 }
 